diff --git a/sound/usb/quirks.c b/sound/usb/quirks.c
index 3ecd1ba7fd4b..36ea5aabd41d 100644
--- a/sound/usb/quirks.c
+++ b/sound/usb/quirks.c
@@ -1390,6 +1390,25 @@ static int snd_usb_motu_m_series_boot_quirk(struct usb_device *dev)
 	return 0;
 }
 
+static int snd_usb_goxlr_boot_quirk(struct usb_device *dev)
+{
+	/* Load the 48000Hz rate into a buffer */
+	u8 srate[4] = {0x80, 0xbb, 0x00, 0x00};
+	u8 temp[24];
+
+	/* Clear the Vendor buffer, we don't need to do anything with this data */
+	snd_usb_ctl_msg(dev, usb_rcvctrlpipe(dev,0), 0x00,
+		USB_DIR_IN | USB_TYPE_VENDOR | USB_RECIP_INTERFACE,
+		0, 0, &temp, 0x0018);
+
+	/* Send the packet to activate the audio at 48000Hz */
+	snd_usb_ctl_msg(dev, usb_sndctrlpipe(dev, 0), 0x01,
+		USB_DIR_OUT | USB_TYPE_CLASS | USB_RECIP_INTERFACE,
+		0x0100, 0x2900, &srate, 0x0004);
+
+	return 0;
+}
+
 /*
  * Setup quirks
  */
@@ -1617,6 +1636,10 @@ int snd_usb_apply_boot_quirk(struct usb_device *dev,
 		    get_iface_desc(intf->altsetting)->bInterfaceNumber < 3)
 			return snd_usb_motu_microbookii_boot_quirk(dev);
 		break;
+
+	case USB_ID(0x1220, 0x8fe0): /* TC-Helicon GoXLR */
+	case USB_ID(0x1220, 0x8fe4): /* TC-Helicon GoXLR Mini */
+		return snd_usb_goxlr_boot_quirk(dev);
 	}
 
 	return 0;
